.github/workflows/code_checks.yaml

```diff
@@ -11,36 +11,68 @@ jobs:
         runs-on: ubuntu-latest
         strategy:
             matrix:
-                php: ['7.2', '7.3', '7.4']
+                php:
+                    - "7.2"
+                    - "7.3"
+                    - "7.4"
+                symfony:
+                    - 2.*
+                    - 3.*
+                    - 4.*
+                    - 5.*
+                dependency-version:
+                    # - prefer-lowest
+                    - prefer-stable
 
-        name: PHP ${{ matrix.php }} tests
+        name: PHP ${{ matrix.php }} - S ${{ matrix.symfony }} - ${{ matrix.dependency-version }} - tests
         steps:
             # basically git clone
-            -   uses: actions/checkout@v2
+            - uses: actions/checkout@v2
 
-            # use PHP of specific version
-            -   uses: shivammathur/setup-php@v1
-                with:
-                    php-version: ${{ matrix.php }}
-                    coverage: none # disable xdebug, pcov
-            
-            -   run: git --version
-            -   run: git config --global user.email "test@github.com"
-            -   run: git config --global user.name "GitHub Action"
-            -   run: git --version
+            - name: Setup Git
+              run: |
+                  git --version
+                  git config --global user.email "test@github.com"
+                  git config --global user.name "GitHub Action"
+                  git --version
 
-            # if we 2 steps like this, we can better see if composer failed or tests
-            -   run: composer install --no-progress
+            - name: Setup PHP
+              # use PHP of specific version
+              uses: shivammathur/setup-php@v1
+              with:
+                  php-version: ${{ matrix.php }}
+                  coverage: none # disable xdebug, pcov
+                  tools: composer
 
-            -   run: composer tests
+            - name: Install Composer Dependencies
+              # first, remove the ones that are not relevant for the tests, whose dependencies could interfer with
+              # the to-be-re-required symfony versions
+              run: |
+                  composer remove --dev "rector/rector" "symplify/easy-coding-standard" "phpstan/phpstan" "phpstan/phpstan-phpunit" --no-interaction --no-update
```

Comment:
Looks like a package fine tuning.  I would either remove static code analyzers from composer dependencies and move it to a script (composer cmd?!), or just leave packages as is..

Reply:
The "leaving package as is" leads to composer failing due incompatibilities, see e.g. [here](https://github.com/matteosister/GitElephant/runs/578846910). There are many ways to circumvent that error, such as allowing elder rector etc. versions. Removing the dependencies from composer just moves the issue to the static\_tests, where the fine tuning will happen to require them, doesn't it? I will look into package fine tuning and come up with an idea.

Reply:
These are dev tools (static analyzers), so running tests with older php version or packages should be skipped.. 
Another approach I would suggest is to set package version to "*" (or similar with highest version limit..), so these analyzers wouldn't prevent installing dependencies for older php versions, and run such tests on latest php version..
Anyway these are for dev/test only, no need to be strict in versioning.